### PR TITLE
Modify AST for require statements

### PR DESCRIPTION
**Pull request by ansharlubis**
Change from 
```
require R1(T, F)
require R2(T, G)
```

to 
```
require :: R1(T, F), R2(T, G)
```

### CODE REVIEW

**Comment by certik** [comment]
Let's add a test for multiple requires in the same line, as in `require :: R1(T, F), R2(T, G)`. I think it is not tested yet.

**Comment by ansharlubis** [comment]
We already have `template_simple_01.f90` and `template_simple_02.f90` that put multiple requires in the same line.

File: src/lfortran/parser/parser.yy [hunk 1]
```diff
@@ -381,7 +381,10 @@ void yyerror(YYLTYPE *yyloc, LCompilers::LFortran::Parser &p,
 %type <ast> derived_type_decl
 %type <ast> template_decl
 %type <ast> requirement_decl
-%type <ast> requires_decl
+//%type <ast> requires_decl
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/parser/parser.yy:384]: Let's remove this, I think.

File: src/lfortran/parser/parser.yy [hunk 2]
```diff
@@ -1033,7 +1044,8 @@ temp_decl
     : var_decl
     | interface_decl
     | derived_type_decl
-    | requires_decl
+    //| requires_decl
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/parser/parser.yy:1047]: This as well.

File: src/lfortran/parser/parser.yy [hunk 3]
```diff
@@ -1569,7 +1581,8 @@ decl_statement
     | enum_decl
     | statement
     | instantiate
-    | requires_decl
+    | require_decl
+    //| requires_decl
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/parser/parser.yy:1585]: ..

File: tests/reference/asr-template_error_01-b4ca796.stderr [hunk 4]
```diff
@@ -1,12 +1,5 @@
...
-75 |             avg = avg_S_from_T(d1, D_divided_by_T(d1, s1), d2, D_divided_by_S(d2, s2))
-   |                                                       ^^ type 't' expected, but 's' provided
-   |
-45 |            pure function D_divided_by_T(n, d) result(quotient)
-   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
-...
-   |
-50 |    
-   | ...^ function definition has parameter type 't'
+59 |         requires operations(D, T, S, plus_D, plus_T, D_divided_by_T, D_divided_by_S)
```
Comment (Thirumalai-Shaktivel) [inline tests/reference/asr-template_error_01-b4ca796.stderr:4]: This needs to be reverted, I think.
Just replace `requires ..` to `require :: ..` as you did it for integration_tests.
I think these files are in the tests/errors folder.

Do the same for other tests as well.

**Comment by Thirumalai-Shaktivel** [comment]
Otherwise, this looks great!

File: tests/reference/asr-template_error_01-b4ca796.stderr (hunk 4 above)
Comment (ansharlubis) [reply tests/reference/asr-template_error_01-b4ca796.stderr:4]: Oh yeah, I forgot about the errors.

**Review by Thirumalai-Shaktivel — APPROVED** [review]
Thank you!

**Pull request merged**